#goal definition
int32 number
---
#result definition
int32 final_number
---
#feedback
int32 current_number